app/components/post_attachment_opengraph/post_attachment_opengraph.js

```diff
@@ -31,10 +31,30 @@ export default class PostAttachmentOpenGraph extends PureComponent {
         theme: PropTypes.object.isRequired,
     };
 
+    static getDerivedStateFromProps(props, state) {
+        if (props.link !== state.prevLink) {
```

Comment:
Component re-render protection 👍

Reply:
do we need `getDerivedStateFromProps` ? and why?

Reply:
@enahum the use of `getDerivedStateFromProps` in replacement of `componentWillReceiveProps` have been explained by another contributor here: https://github.com/mattermost/mattermost-mobile/pull/3573/files#r347749486 including a nice sample shown here: https://reactjs.org/blog/2018/03/27/update-on-async-rendering.html#side-effects-on-props-change

Reply:
@rahimrahman yes I absolutely understand how it works, that’s exactly my point when I ask if we needed at all.

I can see this component using did update only.

Is about not just replacing what we had. If you can please do explain why is it needed here in this component so I can understand your reasoning and then continue with the review :)

Reply:
Haha, my bad @enahum.  I figured you wanted a deeper explanation.  I'll get back to you on this.  I have some errands to do.

Reply:
@enahum I've taken a bit more time to trace the app itself and this component is being re-rendered after getting response back from redux action getOpenGraphMetadata.  You are right, the getDerivedStateFromProps is not needed.

Reply:
@enahum I've had a busy week since last long weekend.  Anyway, thanks for bringing this up.  I should have spent more time tracing back from the parent to understand how the component is behaving.

I've removed getDerivedStateFromProps.  I've also updated 2 test cases that weren't really reflect how the component was actually used.

Reply:
@enahum @chuttam I will mark this as resolved.